frappe/utils/safe_exec.py

```diff
@@ -191,6 +197,51 @@ def get_safe_globals():
 
 	return out
 
+def safe_enqueue(cmd, queue='default', job_name=None, **kwargs):
+	'''
+		Enqueue method to be executed using a background worker if not already queued
+
+		:param cmd: whitelised method or server script api method
+		:param queue: should be either long, default or short
+		:param job_name: used to identify an enqueue call, used to prevent duplicate calls
+	'''
+
+	job_name = job_name or cmd
+
+	site = frappe.local.site
+	queued_jobs = get_jobs(site=site, queue=queue, key='job_name').get(site) or []
+
+	if job_name not in queued_jobs:
+		return enqueue(
+			'frappe.utils.safe_exec.execute_enqueued_cmd',
+			cmd=cmd, job_name=job_name, queue=queue, **kwargs
+		)
+
+def execute_enqueued_cmd(cmd, **kwargs):
+	'''
+		Executes in a background worker
+		Executes the method if it is a whitelisted method or server script api
+	'''
+	try:
+		# run server script if `cmd` is a server script of type API
+		is_server_script_api = run_server_script_api(cmd, kwargs)
```

Comment:
`execute_cmd` handles both. Why can't we just use that?

Reply:
The reason was to pass `kwargs` separately.

We will have to make use of `form_dict` if we want `execute_cmd` to handle server scripts too

Reply:
Done